Avoid math error when formatting values for display

slinkp opened the pull request:
Zero in a component value will cause MathDomainError when we normalize exponents to multiples of 3 for nice-looking component values in the generated js. Does not seem to be an issue anywhere else, this doesn't indicate a problem with the circuit.  Zero needs no normalizing for display, so we just don't :)

slinkp committed:
Raise a more informative error when component values will cause math error

why-trv commented:
There's this bit of code that prints various warnings if things are wrong with the circuit definition:
https://github.com/why-trv/yet-another-tonestack-calculator/blob/e9298f871e5e4dc0ee6efd6c1a1219fec670655a/resources/lcapy/process-circuit.py#L535-L581

I'd imagine the check for zero values should be somewhere around there (and a warning too). Or am I not getting something?

slinkp commented:
Thanks for the tip! I’ll have a look there and update the PR if/when I get
something better working. Sorry I’ve been busy with other things.


On Sat, Jan 31, 2026 at 12:53 PM why-trv ***@***.***> wrote:

> *why-trv* left a comment (why-trv/yet-another-tonestack-calculator#10)
> <https://github.com/why-trv/yet-another-tonestack-calculator/pull/10#issuecomment-3828923984>
>
> There's this bit of code that prints various warnings if things are wrong
> with the circuit definition:
>
> https://github.com/why-trv/yet-another-tonestack-calculator/blob/e9298f871e5e4dc0ee6efd6c1a1219fec670655a/resources/lcapy/process-circuit.py#L535-L581
>
> I'd imagine the check for zero values should be somewhere around there
> (and a warning too). Or am I not getting something?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/why-trv/yet-another-tonestack-calculator/pull/10#issuecomment-3828923984>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAYCMRYWG74ILR75LYZZXD4JTT3LAVCNFSM6AAAAACJ7E3FESVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTQMRYHEZDGOJYGQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

slinkp committed:
Better more obvious fix for component values of zero

Since the math error only happens when normalizing values for nice display,
we check for zero and just use it as is, since 0 needs no formatting.

slinkp commented:
I had another look at this and what the code is actually doing and had an "aha" moment. Renamed the PR and updated the first comment accordingly. What do you think?

There's no need to throw an exception or warn the user; instead we just happily accept zero and everything proceeds merrily.
